### DISCUSSION TITLE
EventLoopType and asynchrony

### DISCUSSION THREAD

**Discussion by jamesdbrock** [fork-tongue/collagraph#103] [General]
https://github.com/fork-tongue/collagraph/blob/c23e173b615f90ebdbc56a658be518a2576d3381/collagraph/types.py#L6-L9

Can I use Python’s built-in `async`/`await` [coroutines](https://peps.python.org/pep-0492/) with the `QT` event loop?

**Comment by jamesdbrock** [comment]
The only documentation I can find about PySide6 asynchrony is very vague. What’s the best way to do coroutines in __collagraph__?

https://doc.qt.io/qtforpython-6/examples/example_async_minimal.html

https://github.com/fancidev/qtinter

https://github.com/CabbageDevelopment/qasync

**Comment by Korijn** [comment]
It's complicated: https://github.com/harvimt/quamash

Personally I think until Pyside6 supports it natively we won't really be able to do it in an elegant/efficient manner.

Your first link is the best you can do with Qt today. I guess you could try to generalize that AsyncHelper.
I can explain the technical details but there's a lot to unpack here.

**Reply by jamesdbrock** [reply]
What do you think about https://github.com/CabbageDevelopment/qasync ? It says that it is a successor to __quamash__.

**Reply by Korijn** [reply]
I think it's not worth the added complexity and adding a dependency to this project. We can design something ourselves based on that AsyncHelper pattern you referenced which seems like it should perform similarly with a lot less complicated code.
